doc/chapters/filesystem/index.rst

```diff
@@ -0,0 +1,116 @@
+Filesystems
+***********
+
+This chapter documents the behavior and layout of the basic filesystem of the
+containers and where it picks up the different subfolders and filesystems from.
+
+
+Standard filesystem
+===================
+
+Each container has a rootfs that is bind mounted as the containers root file
+system of said container. The process of creating the filesystem and bind
+mounting different parts is spread out over several involved parties.
+
+1. The root filesystem directory is created and the
+   directory is empty on creation.
+2. It is setup by the ``lxc-softwarecontainer.in`` script, which is run by
+   lxc init when the container is created.
+3. The ``lxc-softwarecontainer.in``  creates several basic directories
+   creating a normal unix environment as defined by Filesystem Hierarchy
+   Standard (FHS).
+4. The filesystem is still mostly empty except for directories and some basic
+   files such as ``/etc/passwd`` etc.
+5. LXC then mounts some basic directories defined in the
+   ``libsoftwarecontainer/softwarecontainer.conf`` (copied to
+   ``/usr/share/lxc/conf`` on installation).
+6. On top of this, non standard mountpoints are also used to mount for example
+   the gateway directories etc inside the filesystem. These are defined using
+   BindMountFolderInContainer (API of softwarecontainer-agent, see
+   :ref:`Getting Started <getting-started>`), or mounted by
+   the different gateways.
+
+The basic filesystem looks like this on startup::
+
+	.
+	├── config
+	└── rootfs
+	    ├── bin
+	    ├── dev
+	    ├── etc
+	    │   ├── group
+	    │   ├── machine-id
+	    │   ├── passwd
+	    │   ├── pulse
+	    │   │   └── client.conf
+	    │   └── resolv.conf
+	    ├── gateways
+	    ├── home
+	    ├── lib
+	    ├── lib64
+	    ├── proc
+	    ├── root
+	    ├── sbin
+	    ├── tmp
+	    └── usr
+	       	├── lib
+	       	├── lib64
+	       	├── local
+	       	└── sbin
+
+
+Mountpoints
+===========
+
+On top of the above structure, mount points can be created anywhere inside the
+container filestructure using the BindMountFolderInContainer and gateway
+configurations which can mount separate files inside the container filesystem.
+
+
+Write Buffered Filesystems
+==========================
+
+A write buffer can be enabled on containers where it is known that the
+application may be misbehaving towards the underlying filesystem. This could be
+overly high number of writes, etc leading to an early death of storage devices
```

Comment:
There should be a period after "etc", since it is an abbreviation, not a word, so "etc."